tests/components/abode/test_init.py

```diff
@@ -0,0 +1,65 @@
+"""Tests for the Abode module."""
+from unittest.mock import patch
+
+from homeassistant.components.abode import (
+    DOMAIN as ABODE_DOMAIN,
+    SERVICE_CAPTURE_IMAGE,
+    SERVICE_SETTINGS,
+    SERVICE_TRIGGER_AUTOMATION,
+)
+from homeassistant.components.alarm_control_panel import DOMAIN as ALARM_DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.setup import async_setup_component
+
+from .common import MockConfigEntry, setup_platform
+
+
+async def test_abode_setup_from_config(hass):
+    """Test setup from configuration yaml file."""
+    config = {
+        ABODE_DOMAIN: {
+            CONF_USERNAME: "user@email.com",
+            CONF_PASSWORD: "password",
+            "polling": True,
+        }
+    }
+    response = await async_setup_component(hass, ABODE_DOMAIN, config)
+    assert response
+
+
+async def test_change_settings(hass):
+    """Test change_setting service."""
+    await setup_platform(hass, ALARM_DOMAIN)
+
+    with patch("abodepy.Abode.set_setting") as mock_set_setting:
+        await hass.services.async_call(
+            ABODE_DOMAIN,
+            SERVICE_SETTINGS,
+            {"setting": "confirm_snd", "value": "loud"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        mock_set_setting.assert_called_once()
+
+
+async def test_unload_entry(hass):
+    """Test unloading the Abode entry."""
+    mock_entry = MockConfigEntry(
+        domain=ABODE_DOMAIN,
+        data={CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"},
+    )
+    mock_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(mock_entry.entry_id)
+
+    with patch("abodepy.Abode.logout") as mock_logout, patch(
```

Comment:
Is this patching the correct place?

https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Reply:
I've gone in circles with this one. This is how the `abode.events.stop` and `abode.logout` are being called in the `async def async_unload_entry` function within the abode integration:

````
await hass.async_add_executor_job(hass.data[DOMAIN].abode.events.stop)
await hass.async_add_executor_job(hass.data[DOMAIN].abode.logout)
````

It works as it's currently written with `abodepy.Abode.logout`, and it also works with `homeassistant.components.abode.Abode.logout`. My understanding is latter is the "correct" way to do it but both work. The weird part is `abodepy.event_controller.AbodeEventController.stop` works but if I try `homeassistant.components.abode.Abode.events.stop`, it does not work.

I've noticed a lot of other integrations also just patch the actual third party library directly so that's the route I've taken on all of my patches and it seems to work. I'll keep looking into it.

Reply:
Where to patch depends both on what part of the library we want to patch and how we're importing the library here. If it works I think we can keep it like this for now.